### ISSUE TITLE
[BUG] Could not parse the wp-config.php file

### ISSUE THREAD

**Issue by w33zy** [lucatume/wp-browser#641]
**Environment**
OS: Windows 11
PHP version: 8.2.0 
Installed Codeception version: 5.0.11
Installed wp-browser version: 4.0.3  
WordPress version: 6.4-alpha-56665  
Local development environment: Laragon 6.0
WordPress structure and management: default

**Can you perform the test manually?**
No.

**Codeception configuration file**  
```yaml
namespace: Tests
support_namespace: Support
paths:
    tests: tests
    output: tests/_output
    data: tests/Support/Data
    support: tests/Support
    envs: tests/_envs
actor_suffix: Tester
params:
    - tests/.env
extensions:
    enabled:
        - Codeception\Extension\RunFailed
    config: {  }
    commands:
        - lucatume\WPBrowser\Command\RunOriginal
        - lucatume\WPBrowser\Command\RunAll
        - lucatume\WPBrowser\Command\GenerateWPUnit
        - lucatume\WPBrowser\Command\DbExport
        - lucatume\WPBrowser\Command\DbImport
```

**Env file**
```env
WORDPRESS_ROOT_DIR=D:\Laragon\www\collections
WORDPRESS_DB_URL=mysql://root:@localhost:3306/collections
TEST_TABLE_PREFIX=test_
WORDPRESS_TABLE_PREFIX=wp_
WORDPRESS_URL=https://collections.local
WORDPRESS_DOMAIN=collections.local
WORDPRESS_ADMIN_USER=wzy
WORDPRESS_ADMIN_PASSWORD=uswu44241
CHROMEDRIVER_HOST=localhost
CHROMEDRIVER_PORT=9515
```

**Suite configuration file**
```yaml
# Integration suite configuration
#
# Run integration and "WordPress unit" tests.

actor: EndToEndTester
bootstrap: _bootstrap.php
modules:
    enabled:
        - lucatume\WPBrowser\Module\WPWebDriver
        - lucatume\WPBrowser\Module\WPDb
        - lucatume\WPBrowser\Module\WPFilesystem
        - lucatume\WPBrowser\Module\WPLoader
    config:
        lucatume\WPBrowser\Module\WPWebDriver:
            url: '%WORDPRESS_URL%'
            adminUsername: '%WORDPRESS_ADMIN_USER%'
            adminPassword: '%WORDPRESS_ADMIN_PASSWORD%'
            adminPath: '/wp-admin'
            browser: chrome
            host: '%CHROMEDRIVER_HOST%'
            port: '%CHROMEDRIVER_PORT%'
            window_size: 1200x1000
            capabilities:
              "goog:chromeOptions":
                args:
                  - "--headless"
                  - "--disable-gpu"
                  - "--disable-dev-shm-usage"
                  - "--proxy-server='direct://'"
                  - "--proxy-bypass-list=*"
                  - "--no-sandbox"
        lucatume\WPBrowser\Module\WPDb:
            dbUrl: '%WORDPRESS_DB_URL%'
            dump: 'tests/Support/Data/dump.sql'
            populate: true
            cleanup: true
            reconnect: false
            url: '%WORDPRESS_URL%'
            urlReplacement: false
            tablePrefix: '%WORDPRESS_TABLE_PREFIX%'
        lucatume\WPBrowser\Module\WPFilesystem:
            wpRootFolder: '%WORDPRESS_ROOT_DIR%'
        lucatume\WPBrowser\Module\WPLoader:
            loadOnly: false
            wpRootFolder: "%WORDPRESS_ROOT_DIR%" 
            dbUrl: '%WORDPRESS_DB_URL%'
            domain: '%WORDPRESS_DOMAIN%'
            
# Integration suite configuration
#
# Run integration and "WordPress unit" tests.

actor: IntegrationTester
bootstrap: _bootstrap.php
modules:
    enabled:
        - lucatume\WPBrowser\Module\WPLoader
    config:
        lucatume\WPBrowser\Module\WPLoader:
           wpRootFolder: "%WORDPRESS_ROOT_DIR%" 
           dbUrl: '%WORDPRESS_DB_URL%'
           wpDebug: true
           tablePrefix: '%TEST_TABLE_PREFIX%'
           domain: '%WORDPRESS_DOMAIN%'
           adminEmail: 'admin@%WORDPRESS_DOMAIN%'
           title: 'Integration Tests'
           plugins: ['user-collections/user-collections.php']
           theme: ''
```

**Describe the bug**
On a fresh installation of `wp-browser` run running the `vendor/bin/codecept run` command as per the documentation, I get the follow error in my terminal window.

**Output**
```sh
Codeception PHP Testing Framework v5.0.11 https://helpukrainewin.org

In WPConfigFile.php line 146:

  Could not parse the wp-config.php file:


In WPConfigFile.php line 132:

  [lucatume\WPBrowser\Process\ProcessException (255)]
```

Also the following PHP is added to the error log when the above command is run:
```sh
[23-Sep-2023 18:23:43 UTC] PHP Fatal error:  Uncaught Error: Class "lucatume\WPBrowser\Utils\Serializer" not found in D:\Laragon\www\collections\wp-content\plugins\collections\vendor\lucatume\wp-browser\src\Process\SerializableThrowable.php:27
Stack trace:
#0 D:\Laragon\www\collections\wp-content\plugins\collections\vendor\lucatume\wp-browser\src\Process\Worker\worker-script.php(19): lucatume\WPBrowser\Process\SerializableThrowable->__construct(Object(lucatume\WPBrowser\Process\Protocol\ProtocolException))
#1 {main}
  thrown in D:\Laragon\www\collections\wp-content\plugins\collections\vendor\lucatume\wp-browser\src\Process\SerializableThrowable.php on line 27
```

**To Reproduce**
Follow setup instruction on Window 11

**Expected behavior**
No errors

**Comment by w33zy** [comment]
To add: I just tried a fresh install of WordPress, a test-dummy plugin and wp-browser and I am getting the same issue.
Here is the repo: https://github.com/w33zy/wp-browser-test

I can add the `wp-config.php` file, if needed since this is a local install.

**Comment by lucatume** [comment]
Thanks @w33zy for taking the time to open an issue and provide a repository.
I've reproduced your setup (Laragon) on my Windows machine and could reproduce the issue.
I _think_ it stems from some incorrect handling of Windows paths, I'm looking into it. Further investigation will yield more information.

I will post updates in this issue.

**Comment by w33zy** [comment]
You're welcome.
For what it's worth, I was running v3.1.10 before this.

**Comment by vyarmolenko** [comment]
@lucatume I have the same (or very similar) issue on Linux Mint
```sh
root@a3a427da27d4:/var/www/html/wp-content# codecept run acceptance --debug
Codeception PHP Testing Framework v5.0.11 https://helpukrainewin.org

  [Connecting To Db] {"config":{"tablePrefix":"wp_","populate":true,"cleanup":false,"reconnect":false,"dump":["../../../../../../../../provision/data/dump.sql"],"populator":"MYSQL_PWD="$password" mysql -h $host -u $user $dbname < $dump","urlReplacement":false,"originalUrl":"","waitlock":10,"createIfNotExists":false,"dsn":"mysql:host=mysql;dbname=wordpress_tests","user":"wordpress","password":"wordpress","url":"http://mysite.test"},"options":[]}
  [Db] Connected to default wordpress_tests

In WPConfigFile.php line 147:
                                                     
  [lucatume\WPBrowser\Process\ProcessException (1)]  
  Could not parse the wp-config.php file:            
                                                     

Exception trace:
  at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/WPConfigFile.php:147
 lucatume\WPBrowser\WordPress\WPConfigFile->includeFile() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/WPConfigFile.php:53
 lucatume\WPBrowser\WordPress\WPConfigFile->__construct() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/InstallationState/ConfiguredStateTrait.php:195
 lucatume\WPBrowser\WordPress\InstallationState\Configured->buildConfigured() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/InstallationState/Configured.php:32
 lucatume\WPBrowser\WordPress\InstallationState\Configured->__construct() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/Installation.php:289
 lucatume\WPBrowser\WordPress\Installation->setInstallationState() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/Installation.php:52
 lucatume\WPBrowser\WordPress\Installation->__construct() at /root/.composer/vendor/lucatume/wp-browser/src/Module/WPLoader.php:304
 lucatume\WPBrowser\Module\WPLoader->_initialize() at /root/.composer/vendor/codeception/codeception/src/Codeception/SuiteManager.php:69
 Codeception\SuiteManager->initialize() at /root/.composer/vendor/codeception/codeception/src/Codeception/Codecept.php:256
 Codeception\Codecept->runSuite() at /root/.composer/vendor/codeception/codeception/src/Codeception/Codecept.php:216
 Codeception\Codecept->run() at /root/.composer/vendor/codeception/codeception/src/Codeception/Command/Run.php:646
 Codeception\Command\Run->runSuites() at /root/.composer/vendor/codeception/codeception/src/Codeception/Command/Run.php:467
 Codeception\Command\Run->execute() at /root/.composer/vendor/symfony/console/Command/Command.php:291
 Symfony\Component\Console\Command\Command->run() at /root/.composer/vendor/symfony/console/Application.php:1014
 Symfony\Component\Console\Application->doRunCommand() at /root/.composer/vendor/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at /root/.composer/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /root/.composer/vendor/codeception/codeception/src/Codeception/Application.php:112
 Codeception\Application->run() at /root/.composer/vendor/codeception/codeception/app.php:45
 {closure}() at /root/.composer/vendor/codeception/codeception/app.php:46
 require() at /root/.composer/vendor/codeception/codeception/codecept:7
 include() at /root/.composer/vendor/bin/codecept:119

In WPConfigFile.php line 133:
                                                     
  [lucatume\WPBrowser\Process\ProcessException (1)]  
                                                     

Exception trace:
  at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/WPConfigFile.php:133
 lucatume\WPBrowser\WordPress\WPConfigFile->includeFile() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/WPConfigFile.php:53
 lucatume\WPBrowser\WordPress\WPConfigFile->__construct() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/InstallationState/ConfiguredStateTrait.php:195
 lucatume\WPBrowser\WordPress\InstallationState\Configured->buildConfigured() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/InstallationState/Configured.php:32
 lucatume\WPBrowser\WordPress\InstallationState\Configured->__construct() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/Installation.php:289
 lucatume\WPBrowser\WordPress\Installation->setInstallationState() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/Installation.php:52
 lucatume\WPBrowser\WordPress\Installation->__construct() at /root/.composer/vendor/lucatume/wp-browser/src/Module/WPLoader.php:304
 lucatume\WPBrowser\Module\WPLoader->_initialize() at /root/.composer/vendor/codeception/codeception/src/Codeception/SuiteManager.php:69
 Codeception\SuiteManager->initialize() at /root/.composer/vendor/codeception/codeception/src/Codeception/Codecept.php:256
 Codeception\Codecept->runSuite() at /root/.composer/vendor/codeception/codeception/src/Codeception/Codecept.php:216
 Codeception\Codecept->run() at /root/.composer/vendor/codeception/codeception/src/Codeception/Command/Run.php:646
 Codeception\Command\Run->runSuites() at /root/.composer/vendor/codeception/codeception/src/Codeception/Command/Run.php:467
 Codeception\Command\Run->execute() at /root/.composer/vendor/symfony/console/Command/Command.php:291
 Symfony\Component\Console\Command\Command->run() at /root/.composer/vendor/symfony/console/Application.php:1014
 Symfony\Component\Console\Application->doRunCommand() at /root/.composer/vendor/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at /root/.composer/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /root/.composer/vendor/codeception/codeception/src/Codeception/Application.php:112
 Codeception\Application->run() at /root/.composer/vendor/codeception/codeception/app.php:45
 {closure}() at /root/.composer/vendor/codeception/codeception/app.php:46
 require() at /root/.composer/vendor/codeception/codeception/codecept:7
 include() at /root/.composer/vendor/bin/codecept:119

In wp-config.php line 238:
                                
  [Error]                       
  Undefined constant "ABSPATH"  
                                

Exception trace:
  at /config/mysite/wp-config.php:238
 include() at /root/.composer/vendor/lucatume/wp-browser/src/WordPress/WPConfigFile.php:100
 lucatume\WPBrowser\WordPress\WPConfigFile->toIncludeFile() at closure://function () use ($wpSettingsFile, $wpConfigFile): array {
   1|                   var_dump($wpConfigFile);
 > 2|                return $this->toIncludeFile($wpConfigFile, $wpSettingsFile);
   3|            }:2
 lucatume\WPBrowser\WordPress\WPConfigFile->{closure}() at n/a:n/a
 call_user_func_array() at /root/.composer/vendor/lucatume/wp-browser/includes/opis/closure/src/SerializableClosure.php:109
 lucatume\WPBrowser\Opis\Closure\SerializableClosure->__invoke() at /root/.composer/vendor/lucatume/wp-browser/src/Process/Worker/worker-script.php:16


```

**Comment by lucatume** [comment]
Fixed in 4.0.4.
Let me know if that is not the case.

**Comment by w33zy** [comment]
It is fixed. I can now run the 2 default tests suites but only one works. The integration test suite fails but the end-to-end test suite works.

I will create a new ticket for this issue.
